drivers/usb/device/usb_dc_it82xx2.c

```diff
@@ -0,0 +1,1564 @@
+ *
+ * Calculate the register value written to the EP_FIFO_CTRL which is defined as
+ * ep_fifo_ctrl[reg_idx] here for selecting the corresponding control bit.
+ */
+static uint8_t get_ep_fifo_ctrl_reg_val(uint8_t ep_idx)
+{
+	uint8_t reg_val = (ep_idx < EP8) ?
+		(1 << ep_idx) : (1 << (ep_idx - EP8));
+
+	return reg_val;
+}
+
+/*
+ * Functions epn0n1_ext_ctrl_cfg() and epn0n1_ext_ctrl_cfg_seq_inv() provide
+ * the entrance of configuring the EPN0N1 Extended Ctrl Registers.
+ *
+ * The variable set_clr determines if we set/clear the corresponding bit.
+ */
+static void epn0n1_ext_ctrl_cfg(uint8_t reg_idx, uint8_t bit_mask,
+			bool set_clr)
+{
+	(set_clr) ? (epn0n1_ext_ctrl[reg_idx] |= bit_mask) :
+		(epn0n1_ext_ctrl[reg_idx] &= ~(bit_mask));
+}
+
+static void epn0n1_ext_ctrl_cfg_seq_inv(uint8_t reg_idx, uint8_t bit_mask,
+			bool set_clr)
+{
+	bool check = (set_clr) ?
+		(epn0n1_ext_ctrl[reg_idx] & EPN0_OUTDATA_SEQ) :
+		(epn0n1_ext_ctrl[reg_idx] & EPN1_OUTDATA_SEQ);
+
+	(check) ? (epn0n1_ext_ctrl[reg_idx] &= ~(bit_mask)) :
+		(epn0n1_ext_ctrl[reg_idx] |= bit_mask);
+}
+
+/*
+ * Function epn01n1_check_stall():
+ *
+ * Return the status of STALL bit in the EPN0N1 Extend Control Registers
+ */
+static bool epn01n1_check_stall(uint8_t reg_idx, uint8_t bit_mask)
+{
+	return !!(epn0n1_ext_ctrl[reg_idx] & bit_mask);
+}
+
+/*
+ * The following functions provide the entrance of configuring the
+ * EPN Extended Ctrl Registers.
+ */
+static void epn_ext_ctrl_cfg1(uint8_t reg_idx, uint8_t bit_mask, bool set_clr)
+{
+	(set_clr) ? (epn_ext_ctrl[reg_idx].EPN_EXT_CTRL1 |= bit_mask) :
+		(epn_ext_ctrl[reg_idx].EPN_EXT_CTRL1 &= ~(bit_mask));
+}
+
+static void epn_ext_ctrl_cfg2(uint8_t reg_idx, uint8_t bit_mask, bool set_clr)
+{
+	(set_clr) ? (epn_ext_ctrl[reg_idx].EPN_EXT_CTRL2 |= bit_mask) :
+		(epn_ext_ctrl[reg_idx].EPN_EXT_CTRL2 &= ~(bit_mask));
+}
+
+/* From 98h to 9Dh, the EP45/67/89/1011/1213/1415 Extended Control Registers
+ * are defined, and their bits definitions are as follows:
+ *
+ * Bit    Description
+ *  7     Reserved
+ *  6     EPPOINT5_ISO_ENABLE
+ *  5     EPPOINT5_SEND_STALL
+ *  4     EPPOINT5_OUT_DATA_SEQUENCE
+ *  3     Reserved
+ *  2     EPPOINT4_ISO_ENABLE
+ *  1     EPPOINT4_SEND_STALL
+ *  0     EPPOINT4_OUT_DATA_SEQUENCE
+ *
+ * Apparently, we can tell that the EP4 and EP5 share the same register, and
+ * the EP6 and EP7 share the same one, and the other EPs are defined in the
+ * same way.
+ *
+ * In the function usb_extend_ep_ctrl() we will obtain the mask/flag according
+ * to the bits definitions mentioned above. As for the control code, please
+ * refer to the definition of enum it82xx2_extend_ep_ctrl.
+ */
+static int usb_extend_ep_ctrl(uint8_t ep_idx, enum it82xx2_extend_ep_ctrl ctrl)
+{
+	uint8_t reg_idx, mask;
+	bool flag;
+
+	if (ep_idx < EP4) {
+		return -EINVAL;
+	}
+
+	if ((ctrl >= EXT_EP_DIR_IN) && (ctrl < EXT_EP_READY)) {
+	/* From EXT_EP_DIR_IN to EXT_EP_DISABLE */
```

Comment:
nitpick: comment not aligned

Reply:
Thanks for reminding, sir. I'll fix it in the next upload.